src/Interpreters/InterpreterInsertQuery.h

```diff
@@ -46,21 +46,28 @@ class InterpreterInsertQuery : public IInterpreter, WithContext
         const StorageMetadataPtr & metadata_snapshot,
         const Names & columns,
         ThreadStatusesHolderPtr thread_status_holder = {},
-        std::atomic_uint64_t * elapsed_counter_ms = nullptr);
+        std::atomic_uint64_t * elapsed_counter_ms = nullptr,
+        bool check_access = false);
 
     static void extendQueryLogElemImpl(QueryLogElement & elem, ContextPtr context_);
 
     void extendQueryLogElemImpl(QueryLogElement & elem, const ASTPtr & ast, ContextPtr context_) const override;
 
     StoragePtr getTable(ASTInsertQuery & query);
-    Block getSampleBlock(const ASTInsertQuery & query, const StoragePtr & table, const StorageMetadataPtr & metadata_snapshot) const;
+    static Block getSampleBlock(
+        const ASTInsertQuery & query,
+        const StoragePtr & table,
+        const StorageMetadataPtr & metadata_snapshot,
+        ContextPtr context_,
+        bool no_destination = false,
+        bool allow_materialized = false);
```

Comment:
Why are `no_destination` and `allow_materialized` now also here?

Reply:
Because it's static now